commit 4a53c56a913b002d4f7e23b3379ccb134ed454db
Author: Matthias Clasen <matthias.clasen@gmail.com>
Date:   Thu Aug 19 13:10:49 2010 +0200

    Fix up_polkit_get_subject() D-BUS error return
    
    All callers of up_polkit_get_subject() just exit without returning an error to
    the D-BUS caller if they fail to get the PolicyKit subject. Do that in
    up_polkit_get_subject() itself now, for convenience.
    
    Signed-off-by: Martin Pitt <martin.pitt@ubuntu.com>

diff --git a/src/up-polkit.c b/src/up-polkit.c
index ab2c66d..0ca2433 100644
--- a/src/up-polkit.c
+++ b/src/up-polkit.c
@@ -52,12 +52,19 @@ static gpointer up_polkit_object = NULL;
 PolkitSubject *
 up_polkit_get_subject (UpPolkit *polkit, DBusGMethodInvocation *context)
 {
+	GError *error;
 	const gchar *sender;
 	PolkitSubject *subject;
 
 	sender = dbus_g_method_get_sender (context);
 	subject = polkit_system_bus_name_new (sender);
 
+	if (subject == NULL) {
+		error = g_error_new (UP_DAEMON_ERROR, UP_DAEMON_ERROR_GENERAL, "failed to get PolicyKit subject");
+		dbus_g_method_return_error (context, error);
+		g_error_free (error);
+	}
+
 	return subject;
 }
 
